### ISSUE TITLE
False positive about the rule NP_EQUALS_SHOULD_HANDLE_NULL_ARGUMENT

### ISSUE THREAD

**Issue by LynnBroe** [spotbugs/spotbugs#2558]
Version: 4.7.3

I found a false positive about the rule [NP_EQUALS_SHOULD_HANDLE_NULL_ARGUMENT](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-equals-method-does-not-check-for-null-argument-np-equals-should-handle-null-argument).

In the example below, SpotBugs reports no warning. The `equals` method takes a parameter that is marked as `@javax.annotation.Nonnull` . So it obeys this rule correctly. 

```java
public class Test {
    public boolean equals(@javax.annotation.Nonnull Object a) {
        return a.hashCode() == this.hashCode();
    }
    public boolean something(Test a, Test b) {
        return a.equals(b);
    }
}
```

However, in the example below, SpotBugs reports a warning at line 3 if we use `@edu.umd.cs.findbugs.annotations.NonNull` instead. Actually, `@org.jetbrains.annotations.Nullable` and `@javax.annotation.Nullable` are equivalent. So, this exmaple does not violate the rule NP_EQUALS_SHOULD_HANDLE_NULL_ARGUMENT and the warning is a false positive.

```java
public class Test {
    public boolean equals(@edu.umd.cs.findbugs.annotations.NonNull Object a) { //report a warning, but it is an FP
        return a.hashCode() == this.hashCode();
    }
    public boolean something(Test a, Test b) {
        return a.equals(b);
    }
}

```

**Comment by LynnBroe** [comment]
I find that  `@android.annotation.NonNull` also has the same problem. Please see the code example below:

```java
public class Test {
    public boolean equals(@android.annotation.NonNull Object a) { //report a warning, but it is an FP
        return a.hashCode() == this.hashCode();
    }
    public boolean something(Test a, Test b) {
        return a.equals(b);
    }
}
```

**Comment by JuditKnoll** [comment]
As far as I can see this issue was fixed by https://github.com/spotbugs/spotbugs/pull/2694.
